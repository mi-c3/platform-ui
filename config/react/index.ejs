<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="theme-color" content="#000000">
    <!--
      manifest.json provides metadata used when your web app is added to the
      homescreen on Android. See https://developers.google.com/web/fundamentals/engage-and-retain/web-app-manifest/
    -->
    <link rel="manifest" href="/config/manifest.json">
    <link rel="shortcut icon" href="/favicon.ico">
    <title><%= htmlRspackPlugin.options.title %></title>
    <style type="text/css" media="print">
      * {
          -webkit-print-color-adjust: exact !important; /*Chrome, Safari */
          color-adjust: exact !important;  /*Firefox*/
      }
    </style>
  </head>
  <body>
    <noscript>
      You need to enable JavaScript to run this app.
    </noscript>
    <div id="root">
      <style>
        html, body {
          height: 100%;
          margin: 0;
          padding: 0;
        }
        body {
          overflow: hidden;
          background: linear-gradient(104.04deg, #121940 0%, #182358 51.37%, #0B132C 99.99%);
        }
        .bg-loader {
          background:
          url('/media/login/backgrounds/0'),
          url('images/bg-loader.png');
          background-position: center center;
          background-size: cover;
          background-repeat: no-repeat;
          background-size: cover;
          position: absolute;
          top: 0;
          left: 0;
          right: 0;
          bottom: 0;
          display: flex;
          align-items: center;
          justify-content: center;
          padding: 20px;
        }
        .bg-loader .title {
          font-size: 22px;
          text-transform: uppercase;
          font-family: arial;
          letter-spacing: 4px;
          color: #fff;
          margin-bottom: 8px;
        }
        .bg-loader .logo {
          max-width: 260px;
          width: 100%;
        }
        .bg-loader .bg-content {
          display: flex;
          align-items: center;
          justify-content: center;
          flex-direction: column;
          z-index: 2;
          max-width: 360px;
          width: 100%;
        }
        .bg-loader .loader {
          height: 6px;
          margin-top: 8px;
          border-radius: 8px;
          max-width: 320px;
          width: 100%;
          background:
            linear-gradient(#FFFFFF 0 0),
            linear-gradient(#FFFFFF 0 0),
            #777;
          background-size: 60% 100%;
          background-repeat: no-repeat;
          animation: p 3s infinite;
        }

        @keyframes p {
          0%   {background-position:-150% 0,-150% 0}
          66%  {background-position: 250% 0,-150% 0}
          100% {background-position: 250% 0, 250% 0}
        }
        
        #loader-alert {
          color: #a63e3e;
          background: #f3c0c0;
          border: 1px solid #ffbbbb;
          padding: 12px 24px;
          border-radius: 4px;
          margin-top: 24px;
          font-family: Arial, sans-serif;
          font-size: 14px;
          line-height: 24px;
          display: none;
          text-align: center;
        }
        .image-container {
            width: 100%;
            max-width: 800px;
            aspect-ratio: 2 / 1;
            overflow: hidden;
            display: flex;
            justify-content: center;
            align-items: center;
        }
        .image-container img {
            width: 100%;
            height: 100%;
            object-fit: cover;
        }
      </style>
      <div class="bg-loader">
        <div class="bg-content">
          <h1 class="title">Welcome to</h1>
          <div class="image-container">
          <img class="logo" src="/media/company-logo-l" alt="Company logo" onerror="this.onerror=null; this.src='images/aff-logo-full.png'; this.style.objectFit='contain';" />
          </div>
          <div class="loader">&nbsp;</div>
          <div id="loader-alert">&nbsp;</div>
        </div>
      </div>
    </div>
    <div id="messenger-html"></div>
    <!--
      This HTML file is a template.
      If you open it directly in the browser, you will see an empty page.

      You can add webfonts, meta tags, or analytics to this file.
      The build step will place the bundled scripts into the <body> tag.

      To begin the development, run `npm start`.
      To create a production bundle, use `npm run build`.
    -->
  </body>
</html>
