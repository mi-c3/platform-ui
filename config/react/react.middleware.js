const reactLib = require('./reactLib.middleware');

module.exports = ({ rootDir }) =>
    reactLib({
        rootDir,
        externals: {
            '@date-io/moment': 'commonjs2 @date-io/moment',
            '@material-ui/core': 'commonjs2 @material-ui/core',
            '@material-ui/icons': 'commonjs2 @material-ui/icons',
            '@material-ui/pickers': 'commonjs2 @material-ui/pickers',
            '@material-ui/styles': 'commonjs2 @material-ui/styles',
            'fast-deep-equal': 'commonjs2 fast-deep-equal',
            'google-map-react': 'commonjs2 google-map-react',
            jss: 'commonjs2 jss',
            marked: 'commonjs2 marked',
            'memoize-one': 'commonjs2 memoize-one',
            moment: 'commonjs2 moment',
            react: 'commonjs2 react',
            'react-avatar-editor': 'commonjs2 react-avatar-editor',
            'react-color': 'commonjs2 react-color',
            'react-dom': 'commonjs2 react-dom',
            'react-dropzone': 'commonjs2 react-dropzone',
            'react-jss': 'commonjs2 react-jss',
            'react-mde': 'commonjs2 react-mde',
            'react-router-dom': 'commonjs2 react-router-dom',
            'react-tiny-virtual-list': 'commonjs2 react-tiny-virtual-list',
            'styled-components': 'commonjs2 styled-components',
        },
    });
