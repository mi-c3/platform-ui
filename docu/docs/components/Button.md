---
id: button
title: Button
---

## Notes

Component `Button` based on Material UI `Button`. All properties you can see in [official documentation](https://v3.material-ui.com/api/button/)

## Props

Name     |                          Type                          | Default     | Description
-------- | :----------------------------------------------------: | ----------- | ----------------------------------------------------------------------------------------------
iconName |                         string                         | null        | Icon name for rendering `Button` with `MdiIcon`
color    | **enum**: 'default', 'inherit', 'primary', 'secondary' | 'primary'   | The color of the component. It supports those theme colors that make sense for this component.
variant  |       **enum**: 'text', 'outlined', 'contained'        | 'contained' | Variant of button

## How to use

```javascript
import { Button } from '@mic3/platform-ui';

<Button src={src} initials={initials} />
```

## Storybook

<[Button](/platform-ui/redirect?/storybook/index.html?path=/story/components-button--primary)>
