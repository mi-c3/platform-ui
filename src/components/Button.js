import React from 'react';
import PropTypes from 'prop-types';
import { Button as MuiButton } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import Link from './Link';

import MdiIcon from 'components/MdiIcon';

const useStyles = withStyles({
    icon: {
        marginTop: '2px',
        marginRight: '5px',
        height: '20px',
    },
});

const LinkedButton = ({ to, href, ...restProps }) => {
    return to || href ? <MuiButton {...restProps} component={Link} to={to} href={href} /> : <MuiButton {...restProps} />;
};
LinkedButton.propTypes = {
    to: PropTypes.string,
    href: PropTypes.string,
};

const Button = ({ iconName, iconType, children, classes, color, ...restProps }) =>
    iconName ? (
        <LinkedButton color={color} {...restProps}>
            <MdiIcon size={15} type={iconType || 'mdi'} name={iconName} className={classes.icon} />
            {children}
        </LinkedButton>
    ) : (
        <LinkedButton color={color} {...restProps}>
            {children}
        </LinkedButton>
    );
Button.propTypes = {
    iconType: PropTypes.string,
    iconName: PropTypes.string,
    color: PropTypes.string,
    children: PropTypes.node,
    classes: PropTypes.object,
    to: PropTypes.string,
    href: PropTypes.string,
};

Button.defaultProps = {
    color: 'primary',
    variant: 'contained',
};

export default useStyles(Button); // eslint-disable-line react-hooks/rules-of-hooks
