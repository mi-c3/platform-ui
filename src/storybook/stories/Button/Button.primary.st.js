import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { Button, Grid } from 'index';
import { forButton } from 'storybook/constants/propsVariation';

storiesOf('Components|Button', module)
    .addDecorator(withKnobs)
    .add('Primary', () => {
        const disabled = boolean('Disabled', false, 'Default');
        const disabledOposite = boolean('Disabled Second', true, 'Default');
        const name = text('Name', 'email', 'Default');
        const color = select('Color', forButton.colors, 'primary', 'Style');
        const size = select('Size', forButton.sizes, 'medium', 'Style');
        const variant = select('Variant', forButton.variants, 'contained', 'Style');
        const label = text('Button Text', 'Button', 'Default');
        return (
            <Grid container direction="column" justify="space-between" style={{ height: 150 }}>
                <Grid item container justify="space-around">
                    <Grid item>
                        <Button disabled={disabled} name={name} onClick={action('onclick')} color={color} size={size} variant={variant}>
                            {label}
                        </Button>
                    </Grid>
                    <Grid item>
                        <Button
                            disabled={disabled}
                            name={name}
                            onClick={action('onclick')}
                            color={color}
                            size={size}
                            variant={variant}
                            iconName="checkbox-blank-circle"
                        >
                            {label}
                        </Button>
                    </Grid>
                </Grid>
                <Grid item container justify="space-around">
                    <Grid item>
                        <Button disabled={disabled} name={name} onClick={action('onclick')} color={color} size={size} variant="outlined">
                            {label}
                        </Button>
                    </Grid>
                    <Grid item>
                        <Button
                            disabled={disabled}
                            name={name}
                            onClick={action('onclick')}
                            color={color}
                            size={size}
                            variant="outlined"
                            iconName="checkbox-blank-circle"
                        >
                            {label}
                        </Button>
                    </Grid>
                </Grid>
                <Grid item container justify="space-around">
                    <Grid item>
                        <Button
                            disabled={disabledOposite}
                            name={name}
                            onClick={action('onclick')}
                            color={color}
                            size={size}
                            iconName="checkbox-blank-circle"
                        >
                            {label}
                        </Button>
                    </Grid>
                    <Grid item>
                        <Button
                            disabled={disabledOposite}
                            name={name}
                            onClick={action('onclick')}
                            color={color}
                            size={size}
                            variant="outlined"
                            iconName="checkbox-blank-circle"
                        >
                            {label}
                        </Button>
                    </Grid>
                </Grid>
            </Grid>
        );
    });
