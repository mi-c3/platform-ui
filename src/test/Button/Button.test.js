import React from 'react';
import { render, cleanup, waitForElement, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../../components/Button';

afterEach(cleanup);

it('renders button', async () => {
    const { getByText, getByTestId, container } = render(
        <Button data-testid="btn-primary" variant={'contained'} color={'primary'} size={'medium'}>
            Click Me!
        </Button>
    );
    const btnId = getByTestId('btn-primary');

    await waitForElement(() => getByText('Click Me!'));
    expect(btnId).toHaveTextContent('Click Me!');
    expect(btnId).toHaveClass('MuiButton-containedPrimary-14');
    expect(container.firstChild).toMatchSnapshot();
});

it('calls "onClick" prop on button click', async () => {
    const onClick = jest.fn();
    const { getByText, container } = render(<Button onClick={onClick}>Click Me!</Button>);

    fireEvent.click(getByText('Click Me!'));
    expect(onClick).toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();
});

it('checks button attribute', async () => {
    const { getByTestId, container } = render(
        <Button data-testid="btn-secondary" variant={'contained'} color={'secondary'} disabled={true}>
            Click Me!
        </Button>
    );

    expect(getByTestId('btn-secondary')).toHaveAttribute('disabled');
    expect(container.firstChild).toMatchSnapshot();
});
