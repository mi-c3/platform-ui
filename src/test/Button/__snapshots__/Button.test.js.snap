// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls "onClick" prop on button click 1`] = `
<button
  class="MuiButtonBase-root-65 MuiButton-root-39 MuiButton-contained-50 MuiButton-containedPrimary-51 MuiButton-raised-53 MuiButton-raisedPrimary-54"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label-40"
  >
    Click Me!
  </span>
  <span
    class="MuiTouchRipple-root-68"
  />
</button>
`;

exports[`checks button attribute 1`] = `
<button
  class="MuiButtonBase-root-102 MuiButtonBase-disabled-103 MuiButton-root-76 MuiButton-contained-87 MuiButton-containedSecondary-89 MuiButton-raised-90 MuiButton-raisedSecondary-92 MuiButton-disabled-96"
  data-testid="btn-secondary"
  disabled=""
  tabindex="-1"
  type="button"
>
  <span
    class="MuiButton-label-77"
  >
    Click Me!
  </span>
</button>
`;

exports[`renders button 1`] = `
<button
  class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-contained-13 MuiButton-containedPrimary-14 MuiButton-raised-16 MuiButton-raisedPrimary-17"
  data-testid="btn-primary"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label-3"
  >
    Click Me!
  </span>
  <span
    class="MuiTouchRipple-root-31"
  />
</button>
`;
